src/main/scala/soc/playerRepository/PlayerContext.scala

```diff
@@ -78,6 +99,82 @@ class PlayerContext[GAME <: Inventory[GAME], PLAYER <: Inventory[PLAYER]](val na
+      case BuildCityMove(vertex) =>
+        GameEvent(position, GameAction.BUILD_CITY, Some(ActionSpecification(vertex = vertex.node.toString())))
+      case BuyDevelopmentCardResult(card) => 
+        GameEvent(position, GameAction.BUILD_DEVELOPMENT_CARD)
+      case KnightResult(robber) => 
+        val s = robber.steal.getOrElse(null)
+        val op = if (s == null) -1 else s.victim
+        GameEvent(position, GameAction.ACTIVATE_KNIGHT, Some(ActionSpecification(hex=robber.robberLocation.toString(), otherPlayerPosition=op)))
+      case RoadBuilderMove(road1, road2) => 
+        val r2 = road2.getOrElse(road1) // TODO FIX
+        val edges = Seq(Edge(road1.v1.node.toString(), road1.v2.node.toString()), Edge(r2.v1.node.toString(), r2.v2.node.toString()))
+        GameEvent(position, GameAction.ACTIVATE_ROAD_BUILDING, Some(ActionSpecification(edges = edges)))
+      case YearOfPlentyMove(res1, res2) => 
+        val r1 = resourceMap.getOrElse(res1.name, null)
+        val r2 = resourceMap.getOrElse(res2.name, null)
+        val actionSpec = ActionSpecification(ask=Seq(r1, r2))
+        val result = ActionResult(resourcesTransacted = Map((1, ResourceTransaction(gain = Seq(r1, r2)))))
+        GameEvent(position, GameAction.ACTIVATE_YEAR_OF_PLENTY, Some(actionSpec), Some(result))
+      case MonopolyResult(cardsLost) => 
+        val transacted = cardsLost.map({case (p, resources) => (p, ResourceTransaction(lose=getResources(resources)))})
+        // TODO: This should also include the resource in the action spec
+        GameEvent(position, GameAction.ACTIVATE_MONOPOLY, result = Some(ActionResult(resourcesTransacted = transacted)))
+      case PortTradeMove(give, get) =>
+        val transacted = Map((position, ResourceTransaction(gain = getResources(get), lose = getResources(give))))
+        GameEvent(position, GameAction.PORT_TRADE, result = Some(ActionResult(resourcesTransacted = transacted)))
+      case DiscardResourcesMove(resourceSet) =>
+          // BROKEN: This should not be bundled by player, for now just defualting to the first player 
+          val first = resourceSet.values.head
+          val transacted = Map((position, ResourceTransaction(lose = getResources(first))))
+          GameEvent(position, GameAction.DISCARD, result = Some(ActionResult(resourcesTransacted = transacted)))
+      case EndTurnMove => 
+        GameEvent(position, GameAction.END_TURN)
+    }
+  }
+
+  private def getResources(resourceSet: CatanResourceSet.Resources): Seq[Resource] = {
+    val resources: mutable.ListBuffer[Resource] = mutable.ListBuffer()
+    resourceSet.amountMap.foreach({ case (res, amt) =>
+        for (i <- 0 to amt) {
+          resources += resourceMap.getOrElse(res.name, null)
+        } 
+    })
+    resources.toSeq
+  }
+
+  private def getActionRequestType(request: RequestMessage[GAME, PLAYER]): ActionRequestType = {
```

Comment:
Arguably this could also go in the `GameBehavior`, but I think for now it's better to only send action requests to the actual player who needs to do the action request.